### PR TITLE
TFC Integrations - ServiceNow Service Catalog - version update and Agents Support feature

### PR DESCRIPTION
**Pull request by jastr945**
### What
In preparation for the next release of the ServiceNow Service Catalog app, I'm updating the version number and adding a paragraph explaining how to use the new feature. **We'll merge this PR on the day of the release.**

Changes include the "Overview" and "Service Catalog" pages. Go to Terraform > TFC section > Integrations > ServiceNow.

### Why
I'm working on version 2.2.0 of the ServiceNow Service Catalog integration. The new version introduces a new feature to the app, which exactly mirrors the existing "Execution Mode" functionality in TFC. I've updated the app's version number on the main page and added a paragraph about the new feature.

### Screenshots
This is what the Execution Mode selection will look like in the ServiceNow UI: 
Screenshot 2023-03-16 at 2 59 48 PM

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [N/A] Description links to related pull requests or issues, if any.

#### Content
- [N/A] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [N/A] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [N/A] Pages with related content are updated and link to this content when appropriate.
- [N/A] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [N/A] New pages have metadata (page name and description) at the top.
- [N/A] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [N/A] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [N/A] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/integrations/service-now/service-catalog.mdx [hunk 1]
```diff
@@ -166,3 +166,13 @@ changes are merged to the associated VCS repositories. This is known as the
...
 that all of the ServiceNow workspaces connected to a given repository will be
 updated whenever changes are merged to the associated branch in that repository.
+
+## Execution Mode
+
+Starting from version 2.2.0, the Service Catalog app allows you to set the [execution mode](https://developer.hashicorp.com/terraform/cloud-docs/workspaces/settings#execution-mode) for your Terraform Workspaces.
+
+ - The default value is "Remote", which performs runs on Terraform Cloud's infrastructure. 
+ - Changing the execution mode to "Local" allows you to execute Terraform runs locally using the CLI workflow. 
+ - Selecting "Agent" enables you to run Terraform operations on isolated, private, or on-premises infrastructure. This option requires an Agent Pool to be created in your Terraform Organization beforehand. Provide its ID in the "Agent Pool ID" field when you order a new Workspace through the Service Catalog.
```
Comment (rkoron007) [inline website/docs/cloud-docs/integrations/service-now/service-catalog.mdx:176]: Styling suggestion for clarity:
```suggestion
 - Selecting "Agent" mode allows you to run Terraform operations on isolated, private, or on-premises infrastructure. This option requires you to create an Agent Pool in your organization beforehand, then provide that Agent Pool's id when you order a new workspace through the Service Catalog.
```

**Review by rkoron007 — COMMENTED** [review]
Hello! I left some suggestions for style, please let me know if you have any questions or feedback! ✨

**Comment by jastr945** [comment]
> Hello! I left some suggestions for style, please let me know if you have any questions or feedback! ✨

Thank you so much for reviewing! I've committed suggested changes.

**Review by rkoron007 — APPROVED** [review]
Looks great to me! ✨

**Pull request merged**